x-pack/plugins/global_search_bar/public/components/search_bar.tsx

```diff
@@ -146,8 +146,10 @@ const resultToOption = (
     'data-test-subj': `nav-search-option`,
   };
 
-  if (type === 'application') option.meta = [{ text: categoryLabel }];
-  else option.meta = [{ text: cleanMeta(type) }];
+  option.meta =
+    type === 'application'
+      ? [{ text: categoryLabel }]
+      : [{ text: cleanMeta((meta.displayName as string) ?? type) }];
```

Comment:
Would you mind adding a test for this change?

Reply:
Will probably need to extract some logic out of the component, but that will not hurt anyway, because that thing is a mess actually. Will do